### DISCUSSION TITLE
Any way to work with the unstable Pytorch-Nightly?

### DISCUSSION THREAD

**Discussion by kfxw** [pyg-team/pytorch_geometric#3225] [Q&A]
Hello!

Although I know there is no official stable release of torch-geometric to work with the latest pytorch-nightly, I'm still curious whether there is anything I can do by myself to make torch-geometric/scatter/sparse capable with the unstable pytorch-nightly.

It is awkward sometimes when you really need some experimental features in the unstable pytorch while you are using pyg to train your graph network. But in theory there should be no conflicts.

Thanks!

**Comment by rusty1s** [comment]
Currently, the only way is to compile the extension packages by yourself. Will need to think about how we can support nightly builds as well.

**Reply by kfxw** [reply]
Alright. I'll try to compile it and update later. Thanks!

**Reply by kfxw** [reply]
For anyone who may have the same issue, here I list the main steps to compile from source,

0. Suppose you have your pytorch-nightly installed.

1. If you are using conda, then you need first install the cudatoolkit that has an nvcc compiler inside. This means that the default conda channels like 'nvidia', 'anaconda', 'conda-forge' are not going to work. You can install via the 'hcc' channel, or use any other solution to install nvcc.
```
conda install -c hcc cudatoolkit=<a specified version>
```
2.  Clone git repos of pytorch_scatter, pytorch_sparse, pytorch_cluster, pytorch_spline_conv and pytorch_geometric. Install them one by one with pip. Install pytorch_geometric at last. 
```
# for example, to install pytorch_scatter
git clone https://github.com/rusty1s/pytorch_scatter.git
cd pytorch_scatter
sudo pip install ./
# repeat this for other packages
```

Then you've got it.
